sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundle.java

```diff
@@ -0,0 +1,289 @@
+    if (StringUtils.isNotBlank(config.getConnectionString())) {
+      connectionString = config.getConnectionString();
+    } else {
+      connectionString =
+          String.format(
+              "mongodb://%s:%s@%s/%s",
+              config.getUsername(), config.getPassword(), config.getHosts(), config.getDatabase());
+      if (StringUtils.isNotBlank(config.getOptions())) {
+        connectionString += "?" + config.getOptions();
+      }
+    }
+
+    var cs = new ConnectionString(connectionString);
+    this.database = cs.getDatabase();
+    mongoClient = MongoClients.create(cs);
+
+    registerHealthCheck(environment.healthChecks(), mongoClient, this.database);
+  }
+
+  /**
+   * registers a health check for the mongo database
+   *
+   * @param healthCheckRegistry registry where to register health checks
+   * @param database database name that is used within health check
+   */
+  private void registerHealthCheck(
+      HealthCheckRegistry healthCheckRegistry, MongoClient mongoClient, String database) {
+    healthCheckRegistry.register(
+        "mongo", new MongoHealthCheck(mongoClient.getDatabase((database))));
+  }
+
+  /**
+   * @return the {@link MongoOperations} that can be used to interact with your MongoDB instance
+   */
+  public MongoOperations getMongoOperations() {
+    if (mongoOperations == null) {
+      mongoOperations = createMongoOperations();
+    }
+    return mongoOperations;
+  }
+
+  /**
+   * @return a new Spring Data Mongo repository
+   * @param <S> the type of your entity type
+   * @param <ID> the type of your primary key
+   */
+  public <T extends Repository<S, ID>, S, ID extends Serializable> T createRepository(
+      Class<T> clazz) {
+    MongoRepositoryFactoryBean<T, S, ID> factoryBean = new MongoRepositoryFactoryBean<>(clazz);
+    factoryBean.setMongoOperations(getMongoOperations());
+    factoryBean.afterPropertiesSet();
+    return factoryBean.getObject();
+  }
+
+  private MongoOperations createMongoOperations() {
+    SimpleMongoClientDatabaseFactory mongoDbFactory =
+        new SimpleMongoClientDatabaseFactory(mongoClient, this.database);
+    MongoConverter mongoConverter = getDefaultMongoConverter(mongoDbFactory, getConverters());
+    return new MongoTemplate(mongoDbFactory, mongoConverter);
+  }
+
+  private List<?> getConverters() {
+    List<Converter<?, ?>> converters = new ArrayList<>(DEFAULT_CONVERTERS);
+    converters.addAll(customConverters);
+    return converters;
+  }
+
+  private SpringDataMongoBundle<C> addCustomConverter(Converter<?, ?> converter) {
+    this.customConverters.add(converter);
+    return this;
+  }
+
+  private SpringDataMongoBundle<C> addCustomConverters(Converter<?, ?>... converters) {
+    Arrays.stream(converters).forEach(this::addCustomConverter);
+    return this;
+  }
+
+  private SpringDataMongoBundle<C> addCustomConverters(Collection<Converter<?, ?>> converters) {
+    converters.forEach(this::addCustomConverter);
+    return this;
+  }
```

Comment:
Imho we can simplify this:

```suggestion
  private SpringDataMongoBundle<C> addCustomConverters(Collection<Converter<?, ?>> converters) {
    this.customConverters.addAll(converters);
    return this;
  }
```

Reply:
Feel free to change it 👍